src/utils/auto_settings.py

```diff
@@ -31,6 +24,39 @@ def adjust_settings():
         d2_saved_games = input()
     if not os.path.exists(d2_saved_games):
         assert(f"Could not find D2R Saved Games at {d2_saved_games}")
+    return d2_saved_games
+
+
+def backup_settings(config: Config):
+    d2_saved_games = get_d2r_folder(config)
+    if os.path.exists(d2_saved_games + "\\Settings_backup.json"):
+        r = input("Backup file already exist, are you sure you want to overwrite it? [y] to confirm")
+        if r == 'y':
+            shutil.copyfile(d2_saved_games + "\\Settings.json", d2_saved_games + "\\Settings_backup.json")
+            print("Settings backed up successfully.")
+    else:
+        shutil.copyfile(d2_saved_games + "\\Settings.json", d2_saved_games + "\\Settings_backup.json")
+        print("Settings backed up successfully.")
+
+
+def restore_settings_from_backup(config: Config):
+    close_down_d2()
+    d2_saved_games = get_d2r_folder(config)
+    shutil.copyfile(d2_saved_games + "\\Settings_backup.json", d2_saved_games + "\\Settings.json")
```

Comment:
This will fail with a "FileNotFoundError" if the user did not back up a settings file previously. Maybe catch the error and output some info to the user instead and return.

Reply:
This is now handling the non existent file. I personally prefer not code around try/catch so I added a check that the path exists before attempting to copy. Tested locally seems to work fine